src/meta/src/hummock/compaction/tier_compaction_picker.rs

```diff
@@ -0,0 +1,234 @@
+            return None;
+        }
+
+        let (_, select_level_inputs) =
+            self.pickup_l0_files(level0, &mut level_handlers[0], sst_idx);
+        level_handlers[0].add_pending_task(self.compact_task_id, &select_level_inputs);
+        Some(SearchResult {
+            select_level: Level {
+                level_idx: 0,
+                level_type: LevelType::Overlapping as i32,
+                table_infos: select_level_inputs,
+            },
+            target_level: Level {
+                level_idx: 0,
+                level_type: LevelType::Overlapping as i32,
+                table_infos: vec![],
+            },
+            split_ranges: vec![],
+        })
+    }
+
+    fn pickup_l0_files(
+        &self,
+        level0: &Level,
+        level0_handler: &mut LevelHandler,
+        sst_idx: usize,
+    ) -> (KeyRange, Vec<SstableInfo>) {
+        let mut tier_key_range =
+            KeyRange::from(level0.table_infos[sst_idx].key_range.as_ref().unwrap());
+        let mut compaction_bytes = level0.table_infos[sst_idx].file_size;
+        let mut select_tables = vec![level0.table_infos[sst_idx].clone()];
+        for table in level0.table_infos[sst_idx + 1..].iter() {
+            assert!(!level0_handler.is_pending_compact(&table.id));
+            if compaction_bytes >= self.max_compaction_bytes {
+                break;
+            }
+            compaction_bytes += table.file_size;
+            let other_key_range = KeyRange::from(table.key_range.as_ref().unwrap());
+            tier_key_range.full_key_extend(&other_key_range);
+            select_tables.push(table.clone());
+        }
+        (tier_key_range, select_tables)
+    }
+
+    fn select_first_l0_sst(
+        &self,
+        levels: &[Level],
+        level_handlers: &mut [LevelHandler],
+    ) -> Option<usize> {
+        assert_eq!(levels[0].level_idx, 0);
+
+        let mut sst_idx = None;
+        for (idx, table) in levels[0].table_infos.iter().enumerate() {
+            if !level_handlers[0].is_pending_compact(&table.id) {
+                sst_idx = Some(idx);
+                break;
+            }
+        }
+        let first_idle_idx = match sst_idx {
+            Some(idx) => idx,
+            None => return None,
+        };
+        let tier_key_range = KeyRange::from(
+            levels[0].table_infos[first_idle_idx]
+                .key_range
+                .as_ref()
+                .unwrap(),
+        );
+        for i in 0..first_idle_idx {
+            let table = &levels[0].table_infos[i];
+            if tier_key_range.full_key_overlap(&KeyRange::from(table.key_range.as_ref().unwrap())) {
+                return None;
+            }
+        }
+
+        for table in &levels[1].table_infos {
+            if tier_key_range.full_key_overlap(&KeyRange::from(table.key_range.as_ref().unwrap()))
+                && level_handlers[1].is_pending_compact(&table.id)
+            {
+                return None;
+            }
+        }
```

Comment:
sorry but I doubt does it support parallel?

Reply:
Why? The origin code does not support parallel

Reply:
It does.